src/ru/payonline/sdk/PayOnlineUtils.java

```diff
@@ -0,0 +1,109 @@
+package ru.payonline.sdk;
+
+import javax.net.ssl.HttpsURLConnection;
+import java.io.*;
+import java.math.BigInteger;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.*;
+
+public final class PayOnlineUtils {
+
+    public static String getMD5Hash(String str)  {
+        MessageDigest m;
+        try {
+            m = MessageDigest.getInstance("MD5");
+        } catch (NoSuchAlgorithmException e) {
+            return "";
+        }
+
+        m.reset();
+        m.update(str.getBytes());
+
+        byte[] digest = m.digest();
+        BigInteger bigInt = new BigInteger(1,digest);
+        StringBuilder hash = new StringBuilder(bigInt.toString(16));
+
+        while(hash.length() < 32 ){
+            hash.insert(0, "0");
+        }
+
+        return hash.toString();
+    }
+
+    public static boolean stringIsNullOrWhiteSpace(String str) {
```

Comment:
public static boolean isNullOrEmpty(String s) {
    return s == null || s.length() == 0;
}

public static boolean isNullOrWhitespace(String s) {
    return s == null || isWhitespace(s);

}
private static boolean isWhitespace(String s) {
    int length = s.length();
    if (length > 0) {
        for (int i = 0; i < length; i++) {
            if (!Character.isWhitespace(s.charAt(i))) {
                return false;
            }
        }
        return true;
    }
    return false;
}